### ISSUE TITLE
further developing Windows shell setup instructor guidance

### ISSUE THREAD

**Issue by kempjd** [swcarpentry/shell-novice#1024]
In working with students using the shell, Windows setups are always a slight challenge and the landscape of tools continues to evolve.

I recommend refreshing the overview of Windows tools to be more current, especially as instructors are advised to test them in advance of teaching and stay on top of this changing landscape.

For example:

- msysGit has been superseded by Git for Windows;

- the Software Carpentries Windows Installer has utility but seems less accessible for beginners and does not appears as actively developed;

- alternative and unadvertised tools like MobaXterm have obvious shortcomings (limiting license, outsized features) but will also be particularly advantageous to some students who plan to continue on to other lessons or Carpentries, and it is also the recommended terminal tool in select Carpentries lessons (High-Performance Computing Carpentry, still in development).

I also recommend suggesting that, when appropriate for their audiences, instructors also consider testing Mac and Linux environments, even though shell functionality within these operating systems seems more routine and stable.  This would help further empower an instructor to deal with setup issues to ensure efficient management of student shell setups and avoid encroachment upon class time.

**Comment by gdevenyi** [comment]
Hi, I don't believe any of the maintainers are windows users, so we're a bit limited in being able to rewrite that part of the lesson.

We welcome contributions to improve things :)

**Comment by cara-a-k** [comment]
I agree things need a refresh on the Windows side, particularly with Win10's Windows subsystem for Linux (WSL). I'm both a Windows and Linux user and I'd strongly recommend moving to WSL. I'm happy to write up installation instructions for WSL if the community is willing to move to using it as the default bash option for Windows. The only tricky thing will be the separation of the Linux and Windows file-systems, but that just means we have to add a step where they cd to their Windows home directory.

I think MobaXterm is a good backup option for Win7 users and people who have installation problems, as it can be run directly from a portable executable. I think MobaXterm and WSL both have a major advantage over git bash: they come with a package manager included. This makes them far more useful to a student over the longer term. If there's opposition to MobaXterm though, we can always fall back to git bash for Win7.

**Comment by gdevenyi** [comment]
I'm happy with a transition to MobaXterm + WSL. I hope there's additional folks to test this as I don't have a windows system to verify anything.

I'm very open to a PR around any windows things.

**Comment by gcapes** [comment]
This refers to http://swcarpentry.github.io/shell-novice/guide/#windows by the looks of things? The setup page is more up to date.

One consideration is that Git for Windows will suffice for both the shell and git lessons. MobaXterm is nice, but I don't think it's straightforward to use in combination with git. That doesn't necessarily have to be a deal breaker though.

**Comment by cara-a-k** [comment]
MobaXterm actually seems to be a lot nicer to use with git than git bash is, especially since version 12:
https://blog.mobatek.net/post/mobaxterm-new-release-12.0/
The one issue I've found is with getting the custom text editor configuration to work, instead of it defaulting to VI. I realise it's probably a deal breaker if I can't figure that out.

BTW I haven't forgotten about my promise to update the docs, just been too busy so far.

**Comment by gdevenyi** [comment]
Good catch @cara-a-k I just checked and git is available now, though you need to download a plugin seperately to get any local shell now which is a *big* downgrade in user-friendlyness.

The installed version of git seems to ignore the standard ways to tell it to launch a different editor...

**Comment by gdevenyi** [comment]
I just had an email correspondence with the mobaxterm guys, the variable that is used first for git is GIT_EDITOR, which they set, rather than using VISUAL/EDITOR, so adjusting those doesn't work. I have recommended to them to default to a more user-friendly default editor, and to follow the standard of using VISUAL/EDITOR to set defaults, rather than modifying the GIT_EDITOR variable. I don't have a windows machine to test right now, but you should be able to unset GIT_EDITOR and set EDITOR=nano, VISUAL=nano, after installing nano from the mobaxterm package manager.

**Comment by cara-a-k** [comment]
Cool, thanks for that Gabriel. I just tested it by setting GIT_EDITOR=/bin/mobatexteditor and that worked. Nano isn't installed by default, but we can install it with the built in package manager. 

I'm not sure where you saw that the local terminal is no longer included by default, but it's not the case. So I don't see any barriers to moving forward with MobaXterm. 

I'll get started with MobaXterm instructions first, as it looks like WSL will be a bit more complicated.

**Comment by wirawan0** [comment]
The CygUtils are now a separate plugin, provided here: https://mobaxterm.mobatek.net/plugins.html .

I just found out in a recent (non-Carpentry) workshop that this is not included (for the users who installed the newest MobaXTerm). As a result I could not ask them to launch ssh with complex options.

**Comment by gdevenyi** [comment]
> The CygUtils are now a separate plugin, provided here: mobaxterm.mobatek.net/plugins.html .

This is the issue I ran into, the newest release is in two parts whereas earlier ones were not.

**Comment by cara-a-k** [comment]
I didn't have that issue with 12.4 (released Dec). I can see the CygUtils plugin in the program folder, but I didn't have to download it separately. I guess they realised it was a mistake to not include it by default, and brought it back.

**Comment by cara-a-k** [comment]
![image](https://user-images.githubusercontent.com/13410022/73330110-c5412880-429a-11ea-915b-230369c94086.png)

The plugin comes included with the portable version of 12.4, so I think it's safe to say they decided to make it the default again.
